# UNESCO - Heritage Analysis
---
The purpose of analysis will be clarifying after this line

In [33]:
# import
import pandas as pd

# install and import country_converter
!pip install country_converter
import country_converter as coco


In [34]:
# load data
heritage = pd.read_csv("data/historical-places-analysis.csv")

# summarise data
heritage.info()
print("\nColumn names: ", heritage.columns.to_list())
heritage.head(5)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1248 entries, 0 to 1247
Data columns (total 54 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Name EN                1248 non-null   object 
 1   Name FR                1248 non-null   object 
 2   Name ES                1245 non-null   object 
 3   Name RU                1128 non-null   object 
 4   Name AR                1141 non-null   object 
 5   Name ZH                1081 non-null   object 
 6   Short Description EN   1248 non-null   object 
 7   Short Description FR   1248 non-null   object 
 8   Short Description ES   1223 non-null   object 
 9   Short Description RU   1135 non-null   object 
 10  Short Description AR   1145 non-null   object 
 11  Short Description ZH   1078 non-null   object 
 12  Description EN         1248 non-null   object 
 13  Justification EN       1198 non-null   object 
 14  Date inscribed         1248 non-null   int64  
 15  Seco

,Name EN,Name FR,Name ES,Name RU,Name AR,Name ZH,Short Description EN,Short Description FR,Short Description ES,Short Description RU,...,Main Video Author,Main Video Caption EN,Main Video Caption FR,Main Video Caption ES,Main Video Caption RU,Main Video Caption AR,Main Video Caption ZH,videos_urls,Components,Components Count
0,Minaret and Archaeological Remains of Jam,Minaret et vestiges archéologiques de Djam,Minarete y vestigios arqueológicos de Jam,Минарет и археологические объекты в Джеме,مئذنة جام وبقاياها الاثرية,查姆回教寺院尖塔和考古遗址,"The 65m-tall Minaret of Jam is a graceful, soa...","Haut de 65m, le minaret de Djam est une constr...","Con sus 65 metros de altura, el minarete de Ja...",65-метровый минарет Джема – грациозное и как б...,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,{name: Valley of Hari River including Minaret ...,1
1,Cultural Landscape and Archaeological Remains ...,Paysage culturel et vestiges archéologiques de...,Paisaje cultural y vestigios arqueológicos del...,Культурный ландшафт и археологические находки ...,المنظر الثقافي والبقايا الأثرية في وادي باميان,巴米扬山谷的文化景观和考古遗迹,The cultural landscape and archaeological rema...,Le paysage culturel et les vestiges archéologi...,Este sitio es un exponente de las creaciones a...,Культурный ландшафт и археологические находки ...,...,NHK Nippon Hoso Kyokai,The cultural landscape and archaeological rema...,NaN,NaN,NaN,NaN,NaN,https://www.youtube-nocookie.com/embed/xexPV8m...,{name: Bamiyan Cliff including niches of the 3...,8
2,Historic Centres of Berat and Gjirokastra,Centres historiques de Berat et de Gjirokastra,Ciudad Museo de Gjirokastra,Исторические центры Берата и Джирокастры,المركزان التاريخيان لبيرات وجيروكاسترا,NaN,Berat and Gjirokastra are inscribed as rare ex...,Berat et Gjirokastra sont inscrites en tant qu...,"Situada en el valle del río Drinos, al sur de ...",Исторические центры Берата и Джирокастры являю...,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,"{name: Museum-City of Gjirokastra , ref: 569-0...",2
3,Ancient and Primeval Beech Forests of the Carp...,Forêts primaires et anciennes de hêtres des Ca...,Bosques antiguos y primarios de hayas de los C...,Древние и первобытные буковые леса Карпат и др...,غابات الزان القديمة وغابات الزان البدائية لمنط...,喀尔巴阡山脉及欧洲其它地区的原始山毛榉林,This transnational property includes 93 compon...,"Ce bien transnational, composé de 93 éléments ...",NaN,Расширение территории транснационального серий...,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,"{name: Rrajca, ref: 1133quinquies-013, latitud...",93
4,Djémila,Djémila,Yemila,Древний город Джемила,جميلة,杰米拉,"Situated 900 m above sea-level, Djémila, or Cu...","Djémila, ou Cuicul, avec son forum, ses temple...","Situada a 900 metros sobre el nivel del mar, l...",Расположенный на высоте 900 м над уровнем моря...,...,NHK Nippon Hoso Kyokai,"Situated 900 m above sea-level, Djémila, or Cu...",NaN,NaN,NaN,NaN,NaN,https://www.youtube-nocookie.com/embed/OU_7VBt...,"{name: Djémila, ref: 191, latitude: 36.32056, ...",1


### 1. Mistakes
❌ `cols_to_drop = [["Name " + l for l in langs]]`  means a list wrapped in another <br>
✅ `cols_to_drop = (["Name " + l for l in langs])` mean a singular flat list -- which is what we need
<br><br>
❌ heritage_df_rename = heritage_df.rename(cols_to_rename)<br>
✅ heritage_df_rename = heritage_df.rename(columns=cols_to_rename)

In [35]:
# cleaning
langs = ['FR', 'ES', 'RU', 'AR', 'ZH']
# group it for readability, not tuple()
cols_to_drop = (
    # other langs
    ["Name " + l for l in langs] +
    ["Short Description " + l for l in langs] +
    ["Main Image Caption " + l for l in langs] +
    ["Main Video Caption " + l for l in langs] +

    # unwanted: how do we know which? == back to .info()
    # null or unnecessary
    ["Cultural Criteria", "Natural Criteria", "Main Video Caption EN",
     "Date end", "Danger list", "Main Video", "Main Video Author",
     "videos_urls", "UUID", "ID", "Danger", "Criteria", "Secondary dates",
     "Description EN", "Transboundary", "Components", "Main Image Copyright",
     "Area hectares", "Category ID"
    ]
)
# group cols to rename
cols_to_rename = {
    "Name EN": "Name",
    "Short Description EN": "Description",
    "Justification EN": "Justification",
    "States Names": "Countries",
    "Date inscribed": "Date",
    "Main Image": "Initial Image",
    "Main Image Author": "Image Author",
    "Main Image Caption EN": "Image Caption",
}

# process dropping and renaming
heritage_df = heritage.drop(cols_to_drop, axis=1, errors='ignore')
heritage_df_rename = heritage_df.rename(columns=cols_to_rename)

heritage_df_rename.info()
print("Validate dropping: ", heritage_df.columns.to_list())
print("Validate renaming: ", heritage_df_rename.columns.to_list())
heritage_df_rename.head(15)


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1248 entries, 0 to 1247
Data columns (total 15 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Name              1248 non-null   object
 1   Description       1248 non-null   object
 2   Justification     1198 non-null   object
 3   Date              1248 non-null   int64 
 4   Category          1248 non-null   object
 5   Countries         1248 non-null   object
 6   ISO Codes         1245 non-null   object
 7   Region            1248 non-null   object
 8   Region Code       1248 non-null   object
 9   Initial Image     1236 non-null   object
 10  Image Author      1171 non-null   object
 11  Image Caption     1236 non-null   object
 12  Images            1236 non-null   object
 13  Coordinates       1247 non-null   object
 14  Components Count  1248 non-null   int64 
dtypes: int64(2), object(13)
memory usage: 146.4+ KB
Validate dropping:  ['Name EN', 'Short Description EN', 'J

,Name,Description,Justification,Date,Category,Countries,ISO Codes,Region,Region Code,Initial Image,Image Author,Image Caption,Images,Coordinates,Components Count
0,Minaret and Archaeological Remains of Jam,"The 65m-tall Minaret of Jam is a graceful, soa...","Brief synthesis At 1,900 m above sea level and...",2002,Cultural,Afghanistan,AF,Asia and the Pacific,APA,https://whc.unesco.org/document/119616,Claudio Margottini,Minaret and Archaeological Remains of Jam,"https://whc.unesco.org/document/119616, https:...","34.3964166667, 64.5158888889",1
1,Cultural Landscape and Archaeological Remains ...,The cultural landscape and archaeological rema...,Brief synthesis Enclosed between the high moun...,2003,Cultural,Afghanistan,AF,Asia and the Pacific,APA,https://whc.unesco.org/document/109141,Graciela Gonzalez Brigas,General view of the Bamiyan Valley,"https://whc.unesco.org/document/109140, https:...","34.84694, 67.82525",8
2,Historic Centres of Berat and Gjirokastra,Berat and Gjirokastra are inscribed as rare ex...,These two fortified historic centres are remar...,2005,Cultural,Albania,AL,Europe and North America,EUR,https://whc.unesco.org/document/111179,Anastasia Tzigounaki,Historic Centres of Berat and Gjirokastra (Alb...,"https://whc.unesco.org/document/118688, https:...","40.0741666667, 20.1408333333",2
3,Ancient and Primeval Beech Forests of the Carp...,This transnational property includes 93 compon...,Brief synthesis The “Ancient and Primeval Beec...,2007,Natural,"North Macedonia,Spain,Italy,France,Poland,Alba...","MK, ES, IT, FR, PL, AL, AT, BE, HR, RO, UA, BG...",Europe and North America,EUR,https://whc.unesco.org/document/181193,RNN Forêt de la Massane,Ancient and Primeval Beech Forests of the Carp...,"https://whc.unesco.org/document/187751, https:...","48.9, 22.1833333333",93
4,Djémila,"Situated 900 m above sea-level, Djémila, or Cu...",Brief synthesis The site of Djémila is located...,1982,Cultural,Algeria,DZ,Arab States,ARB,https://whc.unesco.org/document/109038,Yvon Fruneau,"Situated 900 m above sea-level, Djémila, or Cu...",https://whc.unesco.org/document/109038,"36.32056, 5.73667",1
5,Tassili n'Ajjer,Located in a strange lunar landscape of great ...,Brief synthesis Tassili n'Ajjer is a vast plat...,1982,Mixed,Algeria,DZ,Arab States,ARB,https://whc.unesco.org/document/108937,Gruban,Illizi,"https://whc.unesco.org/document/108937, https:...","25.5, 9.0",1
6,Madriu-Perafita-Claror Valley,The cultural landscape of Madriu-Perafita-Clar...,Brief synthesis The Madriu-Perafita-Claror Val...,2004,Cultural,Andorra,AD,Europe and North America,EUR,https://whc.unesco.org/document/115467,Alex Tena,Madriu-Perafita-Claror Valley,"https://whc.unesco.org/document/115457, https:...","42.49472222, 1.595555556",1
7,Los Glaciares National Park,The Los Glaciares National Park is an area of ...,Brief synthesis Los Glaciares National Park is...,1981,Natural,Argentina,AR,Latin America and the Caribbean,LAC,https://whc.unesco.org/document/121618,Philipp Schinz,Los Glaciares National Park (Argentina),"https://whc.unesco.org/document/138950, https:...","-50.0, -73.24944",1
8,Los Alerces National Park,The Los Alerces National Park is located in th...,Brief synthesis Los Alerces National Park is l...,2017,Natural,Argentina,AR,Latin America and the Caribbean,LAC,https://whc.unesco.org/document/158927,Ricardo Villalba,Lake environment with temperate Alerce forest,"https://whc.unesco.org/document/158927, https:...","-42.8528, -71.8728",1
9,Jesuit Missions of the Guaranis: San Ignacio M...,"The ruins of São Miguel das Missões in Brazil,...",Brief synthesis Jesuit Missions of the Guarani...,1983,Cultural,"Brazil,Argentina","BR, AR",Latin America and the Caribbean,LAC,https://whc.unesco.org/document/116737,M & G Therin-Weise,Ruins of the Jesuit reduction San Ignacio Mini...,"https://whc.unesco.org/document/109700, https:...","-28.54333333, -54.26583333",5


#### Explanation
1. `.str.split(",")` splits the string at the comma <br>  →  ["34.3964166667", "64.5158888889"]<br><br>
2. `.str.split(",", expand=True)` get a list in one column<br>
    | expand=False       | expand=True | expand=True |
    |--------------------|-------------|-------------|
    | 0                  | 0           | 1           |
    | ["34.39", "64.51"] | "34.39"     | "64.51"     |
    | ["12.34", "56.78"] | "12.34"     | "56.78"     |
<br>
3. `.astype(float)` converts string into actual numbers
            <br> == `"34.3964166667"  →  34.3964166667`<br><br>
4. `heritage_df_rename[['Latitude', 'Longitude']] = ...` → a, b = [1, 2]
<br><br>

| Step                                 | Col 0               | Col 1                |
|--------------------------------------|---------------------|----------------------|
| Raw string                           | `"34.3964,64.5158"` | —                    |
| After `.str.split(',', expand=True)` | `"34.3964"`         | `"64.5158"`          |
| After `.astype(float)` → renamed     | `Latitude: 34.3964` | `Longitude: 64.5158` |

In [36]:
# separate into long and lat by a comma
heritage_df_rename[['Latitude', 'Longitude']] = (
    heritage_df_rename['Coordinates'].str.split(",", expand=True).astype(float)
)
heritage_df_rename[['Coordinates', 'Latitude', 'Longitude']].head(20)

,Coordinates,Latitude,Longitude
0,"34.3964166667, 64.5158888889",34.396417,64.515889
1,"34.84694, 67.82525",34.846940,67.825250
2,"40.0741666667, 20.1408333333",40.074167,20.140833
3,"48.9, 22.1833333333",48.900000,22.183333
4,"36.32056, 5.73667",36.320560,5.736670
5,"25.5, 9.0",25.500000,9.000000
6,"42.49472222, 1.595555556",42.494722,1.595556
7,"-50.0, -73.24944",-50.000000,-73.249440
8,"-42.8528, -71.8728",-42.852800,-71.872800
9,"-28.54333333, -54.26583333",-28.543333,-54.265833


#### Natures of Pandas: return a new DataFrame
- by the action `heritage_df_rename = heritage_df_rename.explode('Countries')`, assigns back to the original dataframe
- `.explode()` takes a column with lists and create a new row for each item

In [37]:
heritage_df_rename.columns.to_list()

['Name',
 'Description',
 'Justification',
 'Date',
 'Category',
 'Countries',
 'ISO Codes',
 'Region',
 'Region Code',
 'Initial Image',
 'Image Author',
 'Image Caption',
 'Images',
 'Coordinates',
 'Components Count',
 'Latitude',
 'Longitude']

In [38]:
# change the index number
heritage_df_rename.index = range(1, len(heritage_df_rename) + 1)
print("Size of data BEFORE exploding by column 'Countries': ", heritage_df_rename.shape)

# # .explode() expected a list, but we had strings
# # split to string by "," to a list
heritage_df_rename['Countries'] = heritage_df_rename['Countries'].str.split(",")
heritage_df_rename['ISO Codes'] = heritage_df_rename['ISO Codes'].str.split(",")
heritage_df_exploded = heritage_df_rename.explode(['Countries', 'ISO Codes'])

print("Size of data AFTER exploding by column 'Countries': ", heritage_df_exploded.shape, "\n")
heritage_df_exploded.head(5)

Size of data BEFORE exploding by column 'Countries':  (1248, 17)
Size of data AFTER exploding by column 'Countries':  (1354, 17) 



,Name,Description,Justification,Date,Category,Countries,ISO Codes,Region,Region Code,Initial Image,Image Author,Image Caption,Images,Coordinates,Components Count,Latitude,Longitude
1,Minaret and Archaeological Remains of Jam,"The 65m-tall Minaret of Jam is a graceful, soa...","Brief synthesis At 1,900 m above sea level and...",2002,Cultural,Afghanistan,AF,Asia and the Pacific,APA,https://whc.unesco.org/document/119616,Claudio Margottini,Minaret and Archaeological Remains of Jam,"https://whc.unesco.org/document/119616, https:...","34.3964166667, 64.5158888889",1,34.396417,64.515889
2,Cultural Landscape and Archaeological Remains ...,The cultural landscape and archaeological rema...,Brief synthesis Enclosed between the high moun...,2003,Cultural,Afghanistan,AF,Asia and the Pacific,APA,https://whc.unesco.org/document/109141,Graciela Gonzalez Brigas,General view of the Bamiyan Valley,"https://whc.unesco.org/document/109140, https:...","34.84694, 67.82525",8,34.846940,67.825250
3,Historic Centres of Berat and Gjirokastra,Berat and Gjirokastra are inscribed as rare ex...,These two fortified historic centres are remar...,2005,Cultural,Albania,AL,Europe and North America,EUR,https://whc.unesco.org/document/111179,Anastasia Tzigounaki,Historic Centres of Berat and Gjirokastra (Alb...,"https://whc.unesco.org/document/118688, https:...","40.0741666667, 20.1408333333",2,40.074167,20.140833
4,Ancient and Primeval Beech Forests of the Carp...,This transnational property includes 93 compon...,Brief synthesis The “Ancient and Primeval Beec...,2007,Natural,North Macedonia,MK,Europe and North America,EUR,https://whc.unesco.org/document/181193,RNN Forêt de la Massane,Ancient and Primeval Beech Forests of the Carp...,"https://whc.unesco.org/document/187751, https:...","48.9, 22.1833333333",93,48.900000,22.183333
4,Ancient and Primeval Beech Forests of the Carp...,This transnational property includes 93 compon...,Brief synthesis The “Ancient and Primeval Beec...,2007,Natural,Spain,ES,Europe and North America,EUR,https://whc.unesco.org/document/181193,RNN Forêt de la Massane,Ancient and Primeval Beech Forests of the Carp...,"https://whc.unesco.org/document/187751, https:...","48.9, 22.1833333333",93,48.900000,22.183333


In [39]:
# categorise region into continents
heritage_df_exploded['Continent'] = coco.convert(
    heritage_df_exploded['Countries'], to='continent'
)
heritage_df_exploded['Continent'].value_counts()

Continent
Europe     563
Asia       386
America    209
Africa     164
Oceania     32
Name: count, dtype: int64

In [40]:
# add south America
south_america = [
    'AR', 'BO', 'BR', 'CL', 'CO', 'EC',
    'GY', 'PY', 'PE', 'SR', 'UY', 'VE'
]

heritage_df_exploded['Continent'] = [
    'South America' if code in south_america
    else 'North America' if continent == 'America'
    else continent
    for code, continent in zip(heritage_df_exploded['ISO Codes'], heritage_df_exploded['Continent'])
]
heritage_df_exploded['Continent'].value_counts()

Continent
Europe           563
Asia             386
Africa           164
North America    128
South America     81
Oceania           32
Name: count, dtype: int64

In [41]:
# check overview stats
print("Total sites: ", heritage_df_exploded['Name'].nunique())
print("Total countries: ", heritage_df_exploded['Countries'].nunique())
print("Year range: ", heritage_df_exploded['Date'].min(), "-", heritage_df_exploded['Date'].max(), "\n")

# top 10
print("Top 10 Countries by numbers of sites: ")
heritage_df_exploded.groupby('Countries')['Name'].nunique().sort_values(ascending=False).head(10)

Total sites:  1248
Total countries:  171
Year range:  1978 - 2025 

Top 10 Countries by numbers of sites: 


Countries
Italy                                                   61
China                                                   60
Germany                                                 55
France                                                  54
Spain                                                   50
India                                                   44
Mexico                                                  36
United Kingdom of Great Britain and Northern Ireland    35
Russian Federation                                      33
Iran (Islamic Republic of)                              29
Name: Name, dtype: int64

In [42]:
# inscription over time
heritage_df_exploded.groupby('Date')['Name'].nunique().sort_values()

Date
1989     7
2002     9
1978    12
2009    13
1990    16
2006    18
2013    19
2018    19
1992    20
2010    21
2017    21
2016    21
1991    22
1984    22
2007    22
2015    24
1982    24
2005    24
2003    24
2024    24
2011    25
2012    26
2014    26
2025    26
1981    26
1988    27
2008    27
1980    27
1994    28
2019    29
1995    29
1986    29
1983    29
1998    30
1985    30
2001    31
2004    32
1993    33
2021    34
1996    37
1987    41
1997    45
1979    45
2023    45
1999    48
2000    61
Name: Name, dtype: int64

In [43]:
# category
heritage_df_exploded['Category'].value_counts()
heritage_df_exploded.groupby(['Continent', 'Category'])['Name'].nunique().unstack()

Category,Cultural,Mixed,Natural
Continent,,,
Africa,100,6,48
Asia,304,11,60
Europe,432,8,52
North America,77,6,38
Oceania,10,6,16
South America,54,4,23


In [44]:
top10 = (
    heritage_df_exploded
    .groupby('Countries')['Name']
    .nunique()
    .sort_values(ascending=False)
    .head(10)
    .reset_index()
)
top10.columns = ['Country', 'Sites']
top10

,Country,Sites
0,Italy,61
1,China,60
2,Germany,55
3,France,54
4,Spain,50
5,India,44
6,Mexico,36
7,United Kingdom of Great Britain and Northern I...,35
8,Russian Federation,33
9,Iran (Islamic Republic of),29


In [45]:
# clean up long official country names for display
name_fixes = {
    'United Kingdom of Great Britain and Northern Ireland': 'United Kingdom',
    'Iran (Islamic Republic of)': 'Iran',
    'Russian Federation': 'Russia',
    'Bolivia (Plurinational State of)': 'Bolivia',
    'Venezuela (Bolivarian Republic of)': 'Venezuela',
    'Korea, Republic of': 'South Korea',
    "Lao People's Democratic Republic": 'Laos',
    'Syrian Arab Republic': 'Syria',
    'Viet Nam': 'Vietnam',
    'Tanzania, United Republic of': 'Tanzania',
    'Congo': 'Republic of Congo',
    'Democratic Republic of the Congo': 'DR Congo',
    "Côte d'Ivoire": 'Ivory Coast',
    'Netherlands (Kingdom of the)': 'Netherlands',
}

heritage_df_exploded['Countries'] = heritage_df_exploded['Countries'].replace(name_fixes)

# check
heritage_df_exploded['Countries'].value_counts().head(15)

Countries
Italy                       61
China                       60
Germany                     55
France                      54
Spain                       50
India                       44
Mexico                      36
United Kingdom              35
Russia                      33
Iran                        29
Japan                       26
United States of America    26
Brazil                      25
Türkiye                     22
Canada                      22
Name: count, dtype: int64

In [46]:
cols = ['Name', 'Description', 'Category', 'Countries', 'ISO Codes', 
        'Region', 'Continent', 'Date', 'Latitude', 'Longitude']

heritage_df_exploded[cols].to_json(
    'heritage.json', 
    orient='records', 
    indent=2,
    force_ascii=False
)
print("Done!")

Done!


In [47]:
import os
os.getcwd()

'/Users/coachellaaaas/Desktop/Projects/Portfolio/personal-portfolio/projects/unesco-heritage'